csrc/device_lower/pass/index.cpp

```diff
@@ -1816,6 +1821,230 @@ Val* hardCodedIndexGenerationForStMatrix(
   return out;
 }
 
+// Goal: Store (tma_m, tma_n) row-major tile in shared memory using stmatrix
+// (stsm_m_tile, stsm_n_tile)
+//
+// Let shared_memory tile be (tma_m = 64, tma_n = 128).
+// Let StMatrix tile be (stsm_m_tile = 16, stsm_n_tile = 16).
+// Let dtype be fp16 or bf16, so dtype_size is 2B.
```

Comment:
Not for this PR of course, but in the future if we have 4 byte elements, do you think there's any way for us to essentially view the input as having an inner dimension of 2 like the `IterType::VectorComponent` dimension we get using `view(tv, DataType::Half)` just for the purposes of the stmatrix call?